engine/runner/src/main/java/org/enso/runner/Main.java

```diff
@@ -0,0 +1,1413 @@
+    return new TerminalIO(historyFilePath);
+  }
+
+  private static <A> A withProfiling(
+      ProfilingConfig profilingConfig,
+      ExecutionContextExecutor executor,
+      java.util.concurrent.Callable<A> main)
+      throws IOException {
+    var path = profilingConfig.profilingPath();
+    var sampler =
+        path.isDefined() ? OutputStreamSampler.ofFile(path.get().toFile()) : new NoopSampler();
+    sampler.start();
+    profilingConfig
+        .profilingTime()
+        .foreach(timeout -> sampler.scheduleStop(timeout.length(), timeout.unit(), executor));
+    scala.sys.package$.MODULE$.addShutdownHook(
+        () -> {
+          try {
+            sampler.stop();
+          } catch (IOException ex) {
+            logger.info("Error stopping sampler", ex);
+          }
+          return BoxedUnit.UNIT;
+        });
+
+    try {
+      return main.call();
+    } catch (IOException ex) {
+      throw ex;
+    } catch (Exception ex) {
+      throw new IOException(ex);
+    } finally {
+      sampler.stop();
+    }
+  }
+
+  /**
+   * Handles `--server` CLI option
+   *
+   * @param line a CLI line
+   * @param logLevel log level to set for the engine runtime
+   */
+  @SuppressWarnings("deprecation")
+  private void runLanguageServer(CommandLine line, Level logLevel) {
+    var maybeConfig = parseServerOptions(line);
+
+    if (maybeConfig.isLeft()) {
+      var errorMsg = maybeConfig.left().get();
+      System.err.println(errorMsg);
+      throw exitFail();
+    } else {
+      var config = maybeConfig.right().get();
+      LanguageServerApp.run(config, logLevel, line.hasOption(Main.DAEMONIZE_OPTION));
+      throw exitSuccess();
+    }
+  }
+
+  @SuppressWarnings("deprecation")
+  private static Either<String, LanguageServerConfig> parseServerOptions(CommandLine line) {
+    var rootId =
+        scala.Option.apply(line.getOptionValue(ROOT_ID_OPTION))
+            .toRight(() -> "Root id must be provided")
+            .flatMap(id -> catchNonFatal(() -> UUID.fromString(id), "Root must be UUID"));
+    var rootPath =
+        scala.Option.apply(line.getOptionValue(ROOT_PATH_OPTION))
+            .toRight(() -> "Root path must be provided");
+    var interfac =
+        scala.Option.apply(line.getOptionValue(INTERFACE_OPTION)).getOrElse(() -> "127.0.0.1");
+    var rpcPortStr =
+        scala.Option.apply(line.getOptionValue(RPC_PORT_OPTION)).getOrElse(() -> "8080");
+    var rpcPort = catchNonFatal(() -> Integer.valueOf(rpcPortStr), "Port must be integer");
+    var dataPortStr =
+        scala.Option.apply(line.getOptionValue(DATA_PORT_OPTION)).getOrElse(() -> "8081");
+    var dataPort = catchNonFatal(() -> Integer.valueOf(dataPortStr), "Port must be integer");
+    scala.Option<String> secureRpcPortStr =
+        scala.Option.apply(line.getOptionValue(SECURE_RPC_PORT_OPTION))
+            .map(x -> scala.Option.apply(x))
+            .getOrElse(() -> scala.Option.empty());
```

Comment:
That's spurious, no? `Option.apply(someVal)` already handles correctly null values.

Reply:
Gone as of 1d17383